### PR TITLE
Feature/eng 2800 add distance reporting to the output in the basic formatter

### PR DESCRIPTION
**Pull request by davidrijsman**
Distance has been added to basic formatter of solutions.

### CODE REVIEW

File: templates/nextroute/main.go [hunk 1]
```diff
@@ -47,8 +47,8 @@ func solver(
 	}
 	last := solutions.Last()
 
-	output := nextroute.Format(ctx, options, solver, last)
-	output.Statistics.Result.Custom = nextroute.DefaultCustomResultStatistics(last)
+	output := factory.Format(ctx, options, solver, last)
```
Comment (sebastian-quintero) [inline templates/nextroute/main.go:50]: This is a breaking change and we should not introduce it. I don't see the benefit of shielding the format function, it is just statistics calculation and it actually makes sense to keep it public to reference when someone is implementing a custom formatter.

**Review by sebastian-quintero — DISMISSED** [review]
I would rather not introduce this breaking change because if someone uses a new version, their current `nextroute` code stops working. Other than that, introducing distance is a great idea.

File: templates/nextroute/main.go (hunk 1 above)
Comment (davidrijsman) [reply templates/nextroute/main.go:50]: We would only be able to not shield the nextroute.Format as the factory.Format is using non exported structures to calculate the distance. Is that what you would like to export, I exported the nextroute.DefaultCustomResultStatistics for the exact reason you are stating, as an example how to enhance/change the formatter?

**Review by sebastian-quintero — COMMENTED** [review]
@davidrijsman and I chatted this thoroughly and the reasoning is that `nextroute` should not know about `schema` or `factory`. This is a fair point and I agree. However, I'm still against the breaking change. I don't want to keep breaking customers, even if it's just an import change. So, agree to disagree 😸 .

**Comment by sebastian-quintero** [comment]
Feel free to remove my request for changes.

**Review by larsbeck — APPROVED** [review]

**Pull request merged**